core/src/main/scala/kafka/server/metadata/KRaftMetadataCache.scala

```diff
@@ -565,3 +531,97 @@ class KRaftMetadataCache(
   }
 }
 
+object KRaftMetadataCache {
+
+  def toCluster(clusterId: String, image: MetadataImage): Cluster = {
```

Comment:
This result is different with `KraftMetadataCache#toCluster` (the old will filter partition base on `listener`).
If we decide to change this one, maybe we need to document it.